### PR TITLE
Import libpng 1.6.21

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Imports libpng 1.6.21 from git://git.code.sf.net/p/libpng/code,
tag ibpng-1.6.21-master-signed

This library is built as a user-mode lib only and is needed to be able
to support png images used in Trusted UI.

Internal test files and other unused files are removed.

Tested-by: Jens Wiklander <jens.wiklander@linaro.org> (QEMU)
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: ta/mk/ta_dev_kit.mk [hunk 1]
```diff
@@ -50,11 +50,12 @@ endif
 cppflags$(sm) += -I. -I$(ta-dev-kit-dir)/include
 
 libdirs += $(ta-dev-kit-dir)/lib
-libnames += utils utee mpa utils zlib utee
+libnames += utils utee mpa utils zlib png utee
```
Comment (pascal-brand-st-dev) [inline ta/mk/ta_dev_kit.mk]: Should we always compile and add `zlib` and  `png`, even when TUI is not set?

**Comment by pascal-brand-st-dev** [comment]
I think `LICENSE` should refer png.h that contains the complete license terms.

File: ta/mk/ta_dev_kit.mk (hunk 1 above)
Comment (jenswi-linaro) [inline ta/mk/ta_dev_kit.mk]: The advantage is that TAs are always compiled in the same way.

**Comment by jenswi-linaro** [comment]
I haven't modified `LICENSE`, I think we should try to modify as few files as possible

**Comment by pascal-brand-st-dev** [comment]
`Reviewed-by: Pascal Brand <pascal.brand@linaro.org>`

**Comment by jenswi-linaro** [comment]
Rebased and tag applied

**Pull request merged**